build-config/resources/Ballerina.toml

```diff
@@ -10,7 +10,4 @@ icon = "icon.png"
 repository = "https://github.com/ballerina-platform/module-ballerinax-zendesk"
 
 [build-options]
-observabilityIncluded = false
-
-[platform.java17]
-graalvmCompatible = true
```

Comment:
Why have we set `graalvmCompatible` flag to `false` ?

Reply:
@ayeshLK Just to clarify, I haven't set the graalVM compatibility to `false` in here. here I've simply removed the graalvm property, because all pure ballerina connectors (which does not require any native dependencies) should get detected as graalVM compatible by default.

Reply:
AFAIR we have marked other connectors with this flag. @ThisaruGuruge any idea ?

Reply:
> @ayeshLK Just to clarify, I haven't set the graalVM compatibility to `false` in here.
> 
> here I've simply removed the graalvm property, because all pure ballerina connectors (which does not require any native dependencies) should get detected as graalVM compatible by default.

@TharmiganK appreciate if you can verify

Reply:
> AFAIR we have marked other connectors with this flag. @ThisaruGuruge any idea ?

If the platform is any (i.e. there are no Java dependencies), the GraalVM compatibility should be inferred from the dependency graph. So in this case, I don't think we need this. We don't have to provide anything under the `platform.java17` field in the toml, if the package is `platform.any`, as I understand it.

Reply:
If this flag is not set then the GraalVM compatiblity is inferred from the package and its dependencies. So if the package is a pure ballerina package then, it is **GraalVM compatible if and only if the dependencies are GraalVM compatible**. If the dependencies are from `ballerina` platform, then all of them are are GraalVM compatible. In that case this flag is not required.

The standard approach is to mark the ballerina package as GraalVM compatible true/false only when we have java dependencies/native code. If you have pure ballerina package skip this flag so that the bal tool and central will infer it from the dependencies.